lkml/lexer.py

```diff
@@ -62,33 +74,18 @@ def scan(self) -> Tuple[tokens.Token, ...]:
             self.scan_until_token()
             ch = self.peek()
             if ch == "\0":
-                self.tokens.append(tokens.StreamEndToken(self.line_number))
+                self.tokens.append(CH_MAPPING[ch](self.line_number))
                 break
-            elif ch == "{":
-                self.advance()
-                self.tokens.append(tokens.BlockStartToken(self.line_number))
-            elif ch == "}":
-                self.advance()
-                self.tokens.append(tokens.BlockEndToken(self.line_number))
-            elif ch == "[":
-                self.advance()
-                self.tokens.append(tokens.ListStartToken(self.line_number))
-            elif ch == "]":
-                self.advance()
-                self.tokens.append(tokens.ListEndToken(self.line_number))
-            elif ch == ",":
-                self.advance()
-                self.tokens.append(tokens.CommaToken(self.line_number))
-            elif ch == ":":
-                self.advance()
-                self.tokens.append(tokens.ValueToken(self.line_number))
             elif ch == ";":
                 if self.peek_multiple(2) == ";;":
                     self.advance(2)
-                    self.tokens.append(tokens.ExpressionBlockEndToken(self.line_number))
+                    self.tokens.append(CH_MAPPING[ch](self.line_number))
             elif ch == '"':
                 self.advance()
                 self.tokens.append(self.scan_quoted_literal())
+            elif ch in CH_MAPPING.keys():
+                self.advance()
+                self.tokens.append(CH_MAPPING[ch](self.line_number))
```

Comment:
This is a nice consolidation of this logic, thanks!

Reply:
Thanks!